common/web/gesture-recognizer/src/tools/unit-test-resources/src/inputSequenceSimulator.ts

```diff
@@ -0,0 +1,315 @@
+      otherTouches = otherTouches || [];
+
+      let touchEventDict: TouchEventInit = {
+        bubbles: true,
+        touches: [touch, ...otherTouches],
+        changedTouches: [touch],
+      }
+
+      let buildEvent = (type: string, dict: TouchEventInit) => {
+        if(window['TouchEvent'] !== undefined) {
+          // Nothing beats the real thing.
+          return new TouchEvent(type, dict);
+        } else {
+          // Effectively, an internal polyfill.  But, if at ALL possible, use a real version instead!
+          return this.buildSyntheticTouchEvent(type, dict);
+        }
+      }
+
+      switch(state) {
+        case 'start':
+          event = buildEvent('touchstart', touchEventDict);
+          break;
+        case 'move':
+          event = buildEvent('touchmove', touchEventDict);
+          break;
+        case 'end':
+          event = buildEvent('touchend', touchEventDict);
+          break;
+      }
+
+      config.targetRoot.dispatchEvent(event);
+
+      return touch;
+    }
+
+    replayMouseSample(sample: JSONObject<com.keyman.osk.InputSample>, state: string, targetElement?: HTMLElement) {
+      let config = this.controller.recognizer.config;
+
+      let event: MouseEvent;
+      let eventDict: MouseEventInit = {
+        bubbles: true,
+        buttons: state == 'end' ? 0 : 1,
+        ...this.getSampleClientPos(sample)
+      };
+
+      switch(state) {
+        case 'start':
+          event = new MouseEvent('mousedown', eventDict);
+          break;
+        case 'move':
+          event = new MouseEvent('mousemove', eventDict);
+          break;
+        case 'end':
+          event = new MouseEvent('mouseup', eventDict);
+          break;
+      }
+
+      targetElement = targetElement || config.targetRoot;
+
+      // Implies that `targetElement` is the target element; .target cannot be set separately
+      // for mouse events, so we can't use config.mouseEventRoot.  Don't worry, the
+      // listener will still receive the event b/c bubbling.
+      targetElement.dispatchEvent(event);
+    }
+
+    /**
+     * Given a previously-recorded sequence of sampled input coordinates, this function
+     * reproduces their corresponding events and runs them through this instance's linked
+     * gesture-recognizer instance.
+     *
+     * It will re-use the original fixture-layout configuration.
+     * @param sequenceTestSpec A previously-recorded sequence of sampled input coordinates.
+     * @param replayExecutor A middleman function to use when simulating the actual events.
+     * @returns The final sample to be simulated.
+     */
+    private replayCore(sequenceTestSpec: RecordedCoordSequenceSet,
+      replayExecutor: (func: () => void, sample?: com.keyman.osk.InputSample) => void): com.keyman.osk.InputSample {
+      let inputs = sequenceTestSpec.inputs;
+      const config = sequenceTestSpec.config;
+
+      this.controller.layoutConfiguration = new FixtureLayoutConfiguration(config);
+
+      /**
```

Comment:
```suggestion
      /*
```

`/**` is only used for javadoc-style comments, not in-function comments. Here and a few other places in this PR :grin:

Reply:
It's also used by TypeScript Intellisense within VSCode, even if it's just within the same function.  I like having that Intellisense.

![image](https://user-images.githubusercontent.com/25213402/183788456-c2454666-f8a4-478f-972e-7d1d642845bf.png)

Therefore, it's _not_ without merits for code maintenance, especially within longer, more-complex functions.

Reply:
Compare:

![image](https://user-images.githubusercontent.com/25213402/183788603-cc203a3c-3531-4985-90b9-90a23a2152dc.png)

Only the `/**` will register with Intellisense.